src/js/polyfills/webvtt.js

```diff
@@ -692,6 +692,11 @@ function moveBoxToLinePosition(window, styleBox, containerBox, boxPositions, num
         let percentage = 0; // Lowest possible so the first thing we get is better.
         let bestAxis;
 
+        // Prevent empty boxes from entering recursive overlap correction.
+        if (!b.height || !b.width) {
```

Comment:
Just curious, was the issue that because height was not defined, it would sometimes cause the `bottom` property to be not a number when set here and then that would mess with the while loop conditions?
https://github.com/jwplayer/jwplayer/blob/1b99becaacfb93e57fa79ea57c3c439eeb8756c4/src/js/polyfills/webvtt.js#L550

Reply:
That's a way of looking at it. The root cause is that some times we're passed a line of captions from the manifest with no text. As far as responsibility goes: HLS.js parses the manifest, our API updates our list of captions tracks, and captionsRenderer renders them, via the webvtt polyfill. 

This was a tricky decision because which solution to pick hinges on whether or not we want to allow caption tracks to have cues without text. I chose this implementation that way, regardless of what path forward we take on how to handle empty text cues, we'll never enter that infinite recursion scenario.